### PR TITLE
[All] Remove topologyAlgorithms classes

### PR DESCRIPTION
**Pull request by epernod**
link to issue #1538 

to be discussed..

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp [hunk 1]
```diff
@@ -140,6 +140,18 @@ std::map<std::string, ComponentChange> uncreatableComponents = {
     {"SparseGridMultipleTopology", Pluginized("v20.06", "SofaNonUniformFem")},
     {"SparseGridRamificationTopology", Pluginized("v20.06", "SofaNonUniformFem")},
 
+
+    /***********************/
+    // REMOVED SINCE v20.12
+    { "DynamicSparseGridTopologyAlgorithms", Deprecated("v20.12", "v21.06") },
```
Comment (damienmarchal) [inline SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp]: Removed vs Deprecated

Comment (epernod) [reply SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp]: I would like to remove them... do we have to go through the 6 months of deprecated before... ?

**Comment by epernod** [comment]
Regarding last review comments (as I'm not sure to be there for the next review):

- I would like to remove those header directly... do we have to go through the 6 months of deprecated before... ?
- Do you want me to put keep empty headers with the SOFA_DEPRECATED_HEADER macro? 
- I would keep the xxxSetGeometryAlgorithms and not put everything inside xxxSetTopologyXXX because the name Geometry make sense here to indicate we depend on the type of dof (aka mechanicalObject)

Tests and scenes have been updated.

**Pull request merged**